.github/workflows/preview.yml

```diff
@@ -0,0 +1,31 @@
+name: Preview PR
+
+on:
+  workflow_run:
+    workflows: ["Build and publish"]
+    types: 
+      - completed
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+      
+      - name: Find PR
+        if: ${{ github.event.workflow_run.event == 'pull_request' }}
+        run: |
+          PR=$(curl https://api.github.com/search/issues?q=${{ github.event.workflow_run.head_sha }} |
+          grep -Po "(?<=${{ github.event.workflow_run.repository.full_name }}\/pulls\/)\d*" | head -1)
+          echo "PR=$PR" >> $GITHUB_ENV
+          
+      - name: Comment previews
+        uses: peter-evans/create-or-update-comment@v1
```

Comment:
Will this work when running a PR from a forked repo? In this case gh-actions should only have read access, so probably not able to post comments. For possible workaround, see: https://github.com/bugout-dev/locust/blob/1bebcc2cac67e90c1efb09b42c37e91cdae829f9/.github/workflows/locust.yaml

Reply:
Yes, it works because the action is on the repository itself (since it is a workflow run trigger) and does not depend on how the PR shapes the action (otherwise, this PR would receive some comments already), as we tried with @kainino0x (and I tried on my own too).

Reply:
ahh, missed the part about this being triggered on workflow run - makes sense then :)

Reply:
@pjoe Thanks!